src/meta/src/manager/catalog/mod.rs

```diff
@@ -413,14 +413,18 @@ where
     pub async fn start_create_table_procedure(&self, table: &Table) -> MetaResult<()> {
         let core = &mut self.core.lock().await.database;
         let key = (table.database_id, table.schema_id, table.name.clone());
-        if !core.has_table(table) && !core.has_in_progress_creation(&key) {
+
+        if core.has_table(table) {
+            bail!("table already exists");
+        } else if core.has_in_progress_creation(&key) {
+            bail!("table is in creating procedure");
```

Comment:
Is this reachable, or should we panic here?

Reply:
No, user might create relation with duplicate name in different session.

Reply:
You're right. So this is actaully "a table with duplicated name is in creating procedure"?